index.js

```diff
@@ -86,72 +84,57 @@ Collector.prototype._notifyIfFlushed = function(callback, err) {
     setImmediate(callback, err);
 };
 
-Collector.prototype._flushSeries = function(seriesName, points, callback) {
+Collector.prototype._flushPoints = function(points, callback) {
     if (!points || points.length === 0) {
         return;
     }
     var self = this;
 
     // only send N points at a time to avoid making requests too large
-    var spliceIndex;
-    if (self._client.protocol == 'udp:' || self._client.protocol == 'line') {
-      spliceIndex = self.computePointCountToSend(points.map(function (point) {
-        return JSON.stringify(point).length;
-      }), MTU_SIZE);
-    } else {
-      spliceIndex = 50;
-    }
+    var spliceIndex = 50
     var batch = points.slice(0, spliceIndex);
     points = points.slice(spliceIndex);
     var opt = { precision: self._time_precision };
 
     self._flushesInFlight++;
-    self._client.writePoints(seriesName, batch, opt, function(err) {
+    self._client.writePoints(batch, opt)
+      .catch(function(err) {
+        self._flushesInFlight--;
+        // TODO if error put points back to send again?
+        self.emit('error', err);
+        self._notifyIfFlushed(callback, err);
+        return;
+      })
+      .then(function() {
         self._flushesInFlight--;
-        if (err) {
-            // TODO if error put points back to send again?
-            self.emit('error', err);
-            self._notifyIfFlushed(callback, err);
-            return;
-        }
-
         // there are more points to flush out
         if (points.length > 0) {
-            self._flushSeries(seriesName, points);
+            self._flushPoints(points);
         }
         self._notifyIfFlushed(callback);
-    });
+      });
 };
 
 Collector.prototype.flush = function(callback) {
-    var self = this;
-
-    Object.keys(self._series).forEach(function(key) {
-        var series = self._series[key];
-        delete self._series[key];
-        self._flushSeries(key, series, callback);
-    });
-    self._notifyIfFlushed(callback);
-};
-
-Collector.prototype._getSeries = function(name, reset) {
-    var series = this._series[name];
-    if (!series) {
-        series = [];
-        this._series[name] = series;
-    }
-    return series;
+    var points = this._points;
```

Comment:
Previously, on a per-series basis, we were getting the points, emptying the array, and then flushing the points. I preserved that behavior.

I don't see why this wouldn't work. My previous intuition was that flushing can take some time, so if we empty the array after flushing, we'll miss the points collected after calling `_flushPoints` but before emptying the array. But that won't happen since this is all single threaded.